File: src/main/java/com/dev/farmmanager/service/stakeholder/StakeholderService.java
package com.dev.farmmanager.service.stakeholder;

import com.dev.farmmanager.domain.dto.stakeholder.StakeholderOptionDto;
import com.dev.farmmanager.domain.entity.Stakeholder;
import com.dev.farmmanager.domain.payload.stakeholder.StakeholderPayload;
import lombok.NonNull;

import java.util.List;
import java.util.Optional;

public interface StakeholderService {

    List<Stakeholder> findAll();

    List<StakeholderOptionDto> findOptions();

    Optional<Stakeholder> getById(@NonNull Integer id);

    Stakeholder create(@NonNull Integer userId, @NonNull StakeholderPayload payload);

    Stakeholder update(@NonNull Integer id, @NonNull StakeholderPayload payload);

    void delete(@NonNull Integer id);
}

File: frontend/src/services/stakeholder.service.ts
import http from '@/api/http'
import type { StakeholderDto, StakeholderOption, StakeholderPayload } from '@/types/stakeholder'

export const stakeholderService = {
  findAll: () =>
    http.get<StakeholderDto[]>('/farm/stakeholder').then((r) => r.data),

  getById: (id: number) =>
    http.get<StakeholderDto>(`/farm/stakeholder/${id}`).then((r) => r.data),

  create: (payload: StakeholderPayload) =>
    http.post<StakeholderDto>('/farm/stakeholder', payload).then((r) => r.data),

  update: (id: number, payload: StakeholderPayload) =>
    http.put<StakeholderDto>(`/farm/stakeholder/${id}`, payload).then((r) => r.data),

  delete: (id: number) =>
    http.delete(`/farm/stakeholder/${id}`),

  findOptions: () =>
    http.get<StakeholderOption[]>('/farm/stakeholder/options').then((r) => r.data),
}
